--- /scripts/mkcompile_h	2025-06-22 05:55:41.810999224 +0800
+++ /scripts/mkcompile_h	2025-06-23 22:45:01.154380817 +0800
@@ -73,6 +73,12 @@
   echo \#define LINUX_COMPILE_BY \"`echo $LINUX_COMPILE_BY | $UTS_TRUNCATE`\"
   echo \#define LINUX_COMPILE_HOST \"`echo $LINUX_COMPILE_HOST | $UTS_TRUNCATE`\"
 
+  if [ -z "$KBUILD_COMPILER_STRING" ]; then
+   CC_VERSION=$($CC -v 2>&1 | grep ' version ' | sed 's/[[:space:]]*$//')
+  else
+   CC_VERSION="$KBUILD_COMPILER_STRING"
+  fi
+
   LD_VERSION=$($LD -v | head -n1 | sed 's/(compatible with [^)]*)//' \
 		      | sed 's/[[:space:]]*$//')
   printf '#define LINUX_COMPILER "%s"\n' "$CC_VERSION, $LD_VERSION"
