--- /arch/arm64/boot/dts/vendor/qcom/xiaomi-sm8250-common.dtsi	2025-06-22 05:53:37.826999271 +0800
+++ /arch/arm64/boot/dts/vendor/qcom/xiaomi-sm8250-common.dtsi	2025-06-22 05:58:53.830999151 +0800
@@ -134,6 +134,10 @@
 		board-sensor = "VIRTUAL-SENSOR";
 		ambient-sensor = "ABT-SENSOR";
 	};
+	irda_regulator: irda-regulator {
+		compatible = "regulator-dummy";
+		regulator-name = "irda_regulator";
+	};
 };
 
 &vendor {
@@ -745,9 +749,10 @@
 &qupv3_se2_spi {
 	status = "ok";
 	ir-spi@0 {
-		compatible = "ir-spi";
+		compatible = "ir-spi-led";
 		reg = <0>;
 		spi-max-frequency = <19200000>;
+		power-supply = <&irda_regulator>;
 		status = "ok";
 	};
 };
