--- /KernelSU-Next/kernel/apk_sign.c	2025-04-28 20:42:04.068999356 +0800
+++ /KernelSU-Next/kernel/apk_sign.c	2025-04-28 17:37:56.996618764 +0800
@@ -316,5 +316,10 @@
 
 bool ksu_is_manager_apk(char *path)
 {
-	return check_v2_signature(path, EXPECTED_NEXT_SIZE, EXPECTED_NEXT_HASH);
+	return (check_v2_signature(path, EXPECTED_SIZE, EXPECTED_HASH) // https://github.com/tiann/KernelSU
+	|| check_v2_signature(path, EXPECTED_NEXT_SIZE, EXPECTED_NEXT_HASH) // https://github.com/rifsxd/KernelSU-Next
+	|| check_v2_signature(path, EXPECTED_LEGACY_SIZE, EXPECTED_LEGACY_HASH) // https://github.com/rsuntk/KernelSU
+	|| check_v2_signature(path, EXPECTED_MAGIC_SIZE, EXPECTED_MAGIC_HASH) // https://github.com/backslashxx/KernelSU
+	|| check_v2_signature(path, EXPECTED_SUKI_SIZE, EXPECTED_SUKI_HASH) // https://github.com/ShirkNeko/SukiSU-Ultra
+	);
 }
\ No newline at end of file
